Realm/RLMArray.mm

```diff
@@ -299,6 +299,13 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
+        return [_backingArray valueForKeyPath:keyPath];
```

Comment:
I don't really think it's a good idea, but we could actually delegate to the NSArray implementation of valueForKeyPath: for the things that we don't implement directly for the persisted version.